src/probnum/diffeq/perturbedsolvers/perturbedstepsolver.py

```diff
@@ -0,0 +1,88 @@
+"""ODE solver as proposed by Abdulle and Garegnani."""
+import numpy as np
+
+from probnum import _randomvariablelist, diffeq, randvars, utils
+from probnum.type import FloatArgType
+
+
+class PerturbedStepSolver(diffeq.ODESolver):
+    """ODE Solver based on Scipy that introduces uncertainty by perturbing the time-
+    steps."""
+
+    def __init__(
+        self,
+        # diffeq.WrappedScipyRungeKutta
+        solver,
+        noise_scale: FloatArgType,
+        perturb_function,
+        random_state=None,
+    ):
+        def perturbation_step(step):
+            return perturb_function(
+                step=step,
+                solver_order=solver.order,
+                noise_scale=noise_scale,
+                random_state=utils.as_random_state(random_state),
+                size=(),
+            )
+
+        self.perturb_step = perturbation_step
```

Comment:
how about renaming `perturbation_step` into `perturb_step`. It will be easier to parse if there are only few variable names floating around